contracts/timelock/timelock.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2023 The Bestchains Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package timelock
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"time"
+
+	"github.com/bestchains/bestchains-contracts/library/context"
+	"github.com/bestchains/bestchains-contracts/library/timer"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// Event is an operation:
+// whose detail is {Payload},
+// fnv-1 hashed as {Hash},
+// set to be available after {TimeStamp}
+type Event struct {
```

Comment:
Since `Event` in chaincode stands for  tx logging in blockchain  ,we should avoid using`Event` as argument..

Use `Operation` instead